02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,63 @@
+const { people } = require("../data.js");
+
+const getPeople = (req, res) => {
+  res.status(200).json({ succes: true, data: people });
+};
+
+const addPerson = (req, res) => {
+  const { name } = req.body;
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, message: "Please provide a name" });
+  } else {
+    people.push({ id: people.length + 1, name: req.body.name });
+    return res.status(201).json({ success: true, name: req.body.name });
+  }
+};
+
+const getPerson = (req, res) => {
+  const { id } = req.params;
+  const person = people.find((person) => person.id === Number(id));
+
+  if (!person) {
+    return res
+      .status(404)
+      .json({ success: false, message: `Person with id ${id} not found!` });
+  }
+  res.status(200).json({ success: true, data: person });
+};
+
+const updatePerson = (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+
+  const person = people.find((person) => person.id === Number(id));
+
+  if (!person) {
+    return res
+      .status(404)
+      .json({ success: false, message: `Person with id ${id} not found!` });
+  }
+  const newPeople = people.map((person) => {
```

Comment:
What happens if I update the person with ID 5 and then retrieve the same person with the GET endpoint?